### PR TITLE
Document releasing backports

### PR DESCRIPTION
**Pull request by johanstokking**
#### Summary

Document releasing backports

#### Changes

- Add instructions on how to release backports from minor versions

#### Notes for Reviewers

@KrishnaIyer please test while doing `v3.6.3`. The idea is that minor is `v3.6.x` and the latest patch is `v3.6.2`

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [ ] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [x] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -464,7 +464,7 @@ This will compile binaries for all supported platforms, `deb`, `rpm` and Snapcra
 
 Releasing a new version consists of the following steps:
 
-1. Creating a `release/<version>` branch(further, called "release branch") (e.g. `release/3.2.1`).
+1. Creating a `release/<version>` branch off `master` or `backports/<minor>` (further, called "release branch") (e.g. `release/v3.2.1`).
```
Comment (htdvisser) [inline DEVELOPMENT.md]: I thought you wanted `backport/<minor>` since we also have singular `release/<version>`

**Comment by htdvisser** [comment]
Should we also describe the process of when / how we select PRs / commits for backports?

File: DEVELOPMENT.md (hunk 1 above)
Comment (johanstokking) [reply DEVELOPMENT.md]: I don't care really. With `3.6.2` I chose for singular, not knowing there's `backports/*` branch protection setup

**Comment by johanstokking** [comment]
>Should we also describe the process of when / how we select PRs / commits for backports?

I don't think we have something clear to describe at the moment, except for security fixes, but that is already covered by the security policy.

File: DEVELOPMENT.md (hunk 1 above)
Comment (htdvisser) [reply DEVELOPMENT.md]: I think we should keep them consistent. So if we do singular `release/<version>`, we should also do singular `backport/<minor>`.

**Pull request merged**